internal/pkg/template/workload.go

```diff
@@ -389,7 +389,7 @@ type PublishOpts struct {
 // Topic holds information needed to render a SNSTopic in a container definition.
 type Topic struct {
 	Name *string
-	Type string
+	Fifo Fifo
```

Comment:
What do you think of
```suggestion
type Topic struct {
	FIFO *FIFO
}

type FIFO struct {
         ContentBasedDeduplication bool
}	
```

and in CFN template we use
```
{{ if $topic.Fifo  }}
```
instead of `{{ if or $topic.Fifo.Enable $topic.Fifo.ContentBasedDeduplication }}` to check if FIFO is enabled.

Basically, if FIFO is a `nil` pointer, then it is not enabled. O/w, it is enable.

Reply:
Great catch, I am doing the exact same thing for the SQS FIFO right now, then only I thought of doing the same of SNS FIFO. I will change it.